crates/floresta-wire/src/p2p_wire/chain_selector.rs

```diff
@@ -278,6 +272,21 @@ where
         info!("Fork point is arround height={height} hash={hash}");
         // at the end, this variable should hold the last block where they agreed
         let mut fork = 0;
+
+        // Getting the acc for the block on which we landed on
+        let (peer1_acc, peer2_acc) = self
+            .grab_both_peers_version(peer1, peer2, hash, height)
+            .await?;
+
+        // Intializing the agree bool for the block on which we landed on
+        let agree = peer1_acc == peer2_acc;
```

Comment:
Rather than hardcoding the agree bool to be false, it could declared for the block from which we starts on to find the fork.